src/components/Chat/InputMessageV2.tsx

```diff
@@ -119,19 +138,200 @@ const InputMessageV2 = ({
+    if (success) {
+      ImagePicker.openPicker({
+        multiple: true,
+        maxFiles: 20,
+        sortOrder: 'asc',
+        smartAlbums: ['RecentlyAdded', 'UserLibrary']
+      }).then(async (medias) => {
+        const newMedias: any = [];
+        for (const media of medias) {
+          if (media?.mime?.includes('video')) {
+            const thumbnailPath = await getVideoThumbnail(media.path);
+            newMedias.push({
+              type: 'video',
+              path: thumbnailPath,
+              pathVideo: media.path,
+              nameVideo: media.filename,
+              typeVideo: media.mime
+            });
+          } else {
+            const imageCropped = await ImagePicker.openCropper({
+              mediaType: 'photo',
+              path: media.path,
+              width: media.width,
+              height: media.height,
+              cropperChooseText: 'Next',
+              freeStyleCropEnabled: true
+            });
+            newMedias.push({type: 'image', path: imageCropped.path});
+          }
+        }
+
+        handleUploadMedia(newMedias);
+      });
+    } else {
+      openAlertPermission(
+        'We’re not able to access your photos, please adjust your permission settings for BetterSocial.'
+      );
+    }
+  };
+
+  const onOpenCamera = async () => {
+    const {success} = await requestCameraPermission();
+    if (success) {
+      ImagePicker.openCamera({
+        mediaType: 'photo'
+      }).then(async (imageRes) => {
+        const imageCropped = await ImagePicker.openCropper({
+          mediaType: 'photo',
+          path: imageRes.path,
+          width: imageRes.width,
+          height: imageRes.height,
+          cropperChooseText: 'Next',
+          freeStyleCropEnabled: true
+        });
+        handleUploadCamera(imageCropped.path);
+      });
+    } else {
+      openAlertPermission(
+        'We’re not able to access your camera, please adjust your permission settings for BetterSocial.'
+      );
+    }
+  };
+
+  const onOpenFile = async () => {
+    const {success} = await requestExternalStoragePermission();
+    if (success) {
+      DocumentPicker.pickSingle({
+        presentationStyle: 'fullScreen'
+      }).then((pickerResult) => {
+        handleFile(pickerResult);
+      });
+    } else {
+      openAlertPermission(
+        'We’re not able to access your document library, please adjust your permission settings for BetterSocial.'
+      );
+    }
+  };
 
   const onChangeInput = (v) => {
     setText(v);
   };
-  const onSelectEmoji = () => {
-    refEmoji.current.close();
+
+  const onOpenGIF = () => {
+    refAttachment.current.close();
+    setTimeout(() => {
+      refGif.current.open();
+    }, 500);
+  };
+
+  const onCloseGIF = () => {
+    refGif.current.close();
+  };
+
+  const onSelectAttachment = () => {
+    refAttachment.current.open();
   };
 
   const handleSendMessage = () => {
-    onSendButtonClicked(text);
+    onSendButtonClicked(text, []);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This section introduces several new functions for handling media uploads, GIF selection, file handling, and permission requests. While the logic seems sound, there are a few areas that could be improved:

- Error handling: There's no error handling in the async functions `handleUploadMedia`, `handleUploadCamera`, `onOpenMedia`, `onOpenCamera`, and `onOpenFile`. If any of these operations fail (e.g., due to network issues or permission denial), the app might crash or behave unexpectedly. Consider adding try-catch blocks around these operations to handle potential errors gracefully.
- Code duplication: The permission request and alert display logic are duplicated in `onOpenMedia`, `onOpenCamera`, and `onOpenFile`. Consider extracting this common logic into a separate function to adhere to the DRY principle.

Here's an example of how you can refactor the permission request and alert display logic:

```diff
+  const handlePermission = async (requestPermissionFunction, errorMessage) => {
+    const { success } = await requestPermissionFunction();
+    if (!success) {
+      openAlertPermission(errorMessage);
+    }
+    return success;
+  };

-  const onOpenMedia = async () => {
-    const { success } = await requestExternalStoragePermission();
-    if (success) {
+  const onOpenMedia = () => {
+    const success = handlePermission(
+      requestExternalStoragePermission,
+      'We’re not able to access your photos, please adjust your permission settings for BetterSocial.'
+    );
+    if (success) {
      // rest of the code...
    }
  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->